doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,830 @@
+Note that the versions of PostgreSQL distributed with RHEL and Ubuntu are different. At the time of writing, RHEL 7
+distributes PostgreSQL version 9.2 and Ubuntu 14.04 distributes version 9.3; the replication configuration is the same
+for both versions.
+
+
+### 4.1 Setting up TLS certificates for database authentication
+This section describes how to create and set up certificate authentication between the slave and master database instances.
+
+For further details on the certificate authentication, see the
+[PostgreSQL documentation](https://www.postgresql.org/docs/9.2/static/auth-methods.html#AUTH-CERT).
+
+
+1. Generate the Certificate Authority key and a self-signed certificate for the root-of-trust:
+
+   ```
+   openssl req -new -x509 -days 7300 -nodes -sha256 -out ca.crt -keyout ca.key -subj '/O=cluster/CN=CA'
+   ```
+   The subject name does not really matter here. Remember to keep the `ca.key` file in a safe place.
+
+   > Alternatively, an existing internal CA can be used for managing the certificates. A sub-CA should be created as the
+   > database cluster root-of-trust and use that for issuing the slave and master certificates.
+
+2. Generate keys and certificates signed by the CA for each postgresql instance, including the master. Do not use the CA
+   certificate and key as the database certificate and key.
+
+   Generate a key and the Certificate Signing Request for it:
+   ```
+   openssl req -new -nodes -days 7300 -keyout server.key -out server.csr -subj "/O=cluster/CN=<nodename>"
+   ```
+
+    **Note:** The `<nodename>` (the subject common name) will be used for identifying the cluster nodes. For slave nodes,
+    it needs to match the replication user name that is added to the master database and the username that the slave node
+    database uses to connect to the master. For example, in a system with one master and two slaves, the names of the nodes
+    could be `master`, `slave1` and `slave2`. Other parts of the subject name do not matter and can be named as is
+    convenient.
+
+    For more information on adding the replication user name to the master database, see chapter
+   [4.3 Configuring the master instance for replication](#43-configuring-the-master-instance-for-replication).
+    Configuring the username into `recovery.conf` on the slave nodes is detailed in chapter
+   [4.4 Configuring the slave instance for replication](#44-configuring-the-slave-instance-for-replication)).
+
+   Sign the CSR with the CA, creating a certificate:
+
+   ```
+   openssl x509 -req -in server.csr -CAcreateserial -CA ca.crt -CAkey ca.key -days 7300 -out server.crt
+   ```
+   Repeat the above steps for each node.
+
+3. Copy the certificates and keys to the nodes:
+
+   First, prepare a directory for them:
+
+   ```bash
+   sudo mkdir -p -m 0755 /etc/xroad/postgresql
+   sudo chmod o+x /etc/xroad
```

Comment:
The search (x) permission for the directory is needed so that postgresql can access the keys in (/etc/xroad/postgresql). It does make it possible for any local user to access a file in that directory but only if the file name is known and the file is "world accessible". Currently, none of the files are. Therefore, I think the risk is negligible.